src/main/scala/com/amazon/deequ/analyzers/applicability/Applicability.scala

```diff
@@ -31,13 +30,82 @@ private[deequ] case class ApplicabilityResult(
   failures: Seq[(String, Throwable)]
 )
 
+private[deequ] object Applicability {
+
+  private def shouldBeNull(nullable: Boolean): Boolean = {
+    nullable && math.random < 0.01
+  }
+
+  private def nullableIf(nullable: Boolean)(n: java.lang.Number): java.lang.Number = {
```

Comment:
I think this is a bit problematic, as all methods now return java.lang.Number instead of their designated type. Why not simply return a double here and then cast it in the invoking methods?

Reply:
Using `Number` still allows spark to create the correct schema for the random data. Am I missing something else?

Reply:
I did not mean that its incorrect, I just think it makes the code harder to read, to understand and to debug. If the return type is Number, than a method which should return a Byte could return a Double with a compiler error. We should not rely on the ability of the invoking code to correctly cast the result.